src/Tasks/Microsoft.NET.Build.Tasks/ConflictResolution/FileUtilities.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.NET.Build.Tasks.ConflictResolution
+{
+    static partial class FileUtilities
+    {
+        public static Version GetFileVersion(string sourcePath)
+        {
+            var fvi = FileVersionInfo.GetVersionInfo(sourcePath);
+
+            if (fvi != null)
+            {
+                return new Version(fvi.FileMajorPart, fvi.FileMinorPart, fvi.FileBuildPart, fvi.FilePrivatePart);
+            }
+
+            return null;
+        }
+
+        static readonly HashSet<string> s_assemblyExtensions = new HashSet<string>(new[] { ".dll", ".exe", ".winmd" }, StringComparer.OrdinalIgnoreCase);
```

Comment:
You ought to.  Don't go out of your way to prevent it.  I'm sure there's a time in the near future where you'll need to make a concerted effort to ensure it works.